corehq/messaging/scheduling/forms.py

```diff
@@ -2569,10 +2568,20 @@ def json_error():
             from corehq.apps.data_dictionary.util import get_data_dict_props_by_case_type
             user_case_properties = (
                 get_data_dict_props_by_case_type(self.domain, exclude_deprecated=True))[USERCASE_TYPE]
-            if any(property_name not in user_case_properties for property_name in value.keys()):
-                raise json_error()
-        if any(not isinstance(vs, list) and any(not isinstance(v, str) for v in vs) for vs in value.values()):
-            raise json_error()
+            invalid_keys = \
+                [property_name for property_name in value.keys() if property_name not in user_case_properties]
+            if len(invalid_keys) > 0:
+                raise ValidationError(_("Invalid JSON: Keys are not valid user case property names: %s")
+                                      % ", ".join(invalid_keys))
+
+        keys_with_invalid_value = []
+        for property_name in value.keys():
```

Comment:
nit: could just do value.items()

also nice clean up with the validation logic hear, it's much easier to reason about

Reply:
I use `property_name` three lines down to append to `keys_with_invalid_value`

Reply:
right, wouldn't this be equivalent:

```
for property_name, property_values in value.items():
    if not isinstance(property_values, list) or any(not isinstance(v, str) for v in property_values):
        keys_with_invalid_value.append(property_name)
```